mreg_cli/commands/dhcp.py

```diff
@@ -85,41 +101,34 @@ def disassoc(args: argparse.Namespace) -> None:
 
     :param args: argparse.Namespace (name)
     """
-    ipaddress_to_use = None
-    ipobjs = IPAddress.get_by_ip(args.name)
-    if ipobjs:
-        if len(ipobjs) > 1:
-            raise InputFailure(f"IP {args.name} is in use by {len(ipobjs)} hosts.")
-
-        ipaddress_to_use = ipobjs[0]
-
-        if not ipaddress_to_use.macaddress:
-            raise InputFailure(f"IP {args.name} does not have a MAC address associated.")
-
-    if not ipaddress_to_use:
-        host = Host.get_by_any_means_or_raise(args.name)
+    name: str = args.name
 
+    ipaddress = ipaddress_from_ip_arg(name)
+    if not ipaddress:
+        host = Host.get_by_any_means_or_raise(name)
         try:
-            ipaddress_to_use = host.has_ip_with_mac(args.name)
+            ipaddress = host.has_ip_with_mac(name)
         except ValueError:
             pass
 
-        if not ipaddress_to_use:
+        if not ipaddress:
             ips_with_mac = host.ips_with_macaddresses()
 
             if not ips_with_mac:
                 raise InputFailure(
                     f"Host {host} does not have any IP addresses with MAC addresses."
-                )
+                ) from None
 
             if len(ips_with_mac) > 1:
-                raise InputFailure(f"Host {host} has multiple IP addresses with MAC addresses.")
+                raise InputFailure(
+                    f"Host {host} has multiple IP addresses with MAC addresses."
+                ) from None
 
-            ipaddress_to_use = ips_with_mac[0]
+            ipaddress = ips_with_mac[0]
 
-    mac = ipaddress_to_use.macaddress
-    ipaddress_to_use.disassociate_mac()
+    ipaddress.disassociate_mac()
```

Comment:
Nitpick. I always thought it looked weird to first something like `disassociate_mac()` and then fetch it via `.macaddress`.

Reply:
I've removed this assignment altogether. We can access it directly when constructing the message.